client/trino-client/src/main/java/io/trino/client/spooling/SpooledSegmentIterator.java

```diff
@@ -62,7 +62,7 @@ public void load()
         checkState(iterator == null, "Iterator should be unloaded");
         try {
             InputStream stream = closer.register(loader.load(segment)); // close stream when exhausted
-            iterator = decoder.decode(stream, segment.getMetadata());
+            iterator = closer.register(decoder.decode(stream, segment.getMetadata()));
```

Comment:
did you grep for all the places where we do `decoder.decode` to see if this only place?

Reply:
There is one more place in test. Fixed